Make clean + make breaks scanner

Running make clean followed by make does not correctly regenerate the scanner.

```
[nathanielc@slate flux][0]$ tree internal/scanner/
internal/scanner/
├── Makefile
├── scanner.gen.go
├── scanner.go
├── scanner.rl
├── scanner_test.go
├── unicode2ragel.rb
├── unicode2ragel.rb.COPYING
└── unicode.rl

0 directories, 8 files
[nathanielc@slate flux][0]$ make clean
make -C ast clean
make[1]: Entering directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/ast'
make -C asttest clean
make[2]: Entering directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/ast/asttest'
rm -f cmpopts.go
make[2]: Leaving directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/ast/asttest'
make[1]: Leaving directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/ast'
make -C internal/scanner clean
make[1]: Entering directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/internal/scanner'
rm -f scanner.gen.go unicode.rl
make[1]: Leaving directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/internal/scanner'
rm -rf bin
[nathanielc@slate flux][0]$ tree internal/scanner/
internal/scanner/
├── Makefile
├── scanner.go
├── scanner.rl
├── scanner_test.go
├── unicode2ragel.rb
└── unicode2ragel.rb.COPYING

0 directories, 6 files
[nathanielc@slate flux][0]$ make
env GO111MODULE=on go build -tags '' -o bin/linux/cmpgen ./ast/asttest/cmpgen
make -C ast
make[1]: Entering directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/ast'
make -C asttest
make[2]: Entering directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/ast/asttest'
PATH=../../bin/linux:/home/nathanielc/.opam/system/bin:/home/nathanielc/.nvm/versions/node/v9.11.2/bin:/home/nathanielc/projects/platform/src/github.com/influxdata/idpe/bin/linux:/home/nathanielc/projects/platform/src/github.com/influxdata/platform/bin/linux:/home/nathanielc/projects/platform/bin:/home/nathanielc/.opam/system/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/smlnj/bin:/home/nathanielc/.gem/ruby/2.2.0/bin:/usr/local/go/bin:/home/nathanielc/bin:/home/nathanielc/.local/bin:/usr/lib/smlnj/bin:/home/nathanielc/gocode/bin:/home/nathanielc/.cargo/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/smlnj/bin:/home/nathanielc/.gem/ruby/2.2.0/bin:/usr/local/go/bin:/home/nathanielc/bin:/home/nathanielc/.local/bin:/usr/lib/smlnj/bin:/home/nathanielc/gocode/bin:/home/nathanielc/.cargo/bin go generate -tags '' -x ./...
cmpgen cmpopts.go
make[2]: Leaving directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/ast/asttest'
make[1]: Leaving directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/ast'
make -C internal/scanner
make[1]: Entering directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/internal/scanner'
make[1]: *** No rule to make target 'unicode.rl', needed by 'scanner.gen.go'.  Stop.
make[1]: Leaving directory '/home/nathanielc/projects/platform/src/github.com/influxdata/flux/internal/scanner'
make: *** [Makefile:33: internal/scanner] Error 2
[nathanielc@slate flux][2]$ tree internal/scanner/
internal/scanner/
├── Makefile
├── scanner.go
├── scanner.rl
├── scanner_test.go
├── unicode2ragel.rb
└── unicode2ragel.rb.COPYING

0 directories, 6 files
[nathanielc@slate flux][0]$ go generate ./internal/scanner/
[nathanielc@slate flux][0]$ tree internal/scanner/
internal/scanner/
├── Makefile
├── scanner.gen.go
├── scanner.go
├── scanner.rl
├── scanner_test.go
├── unicode2ragel.rb
├── unicode2ragel.rb.COPYING
└── unicode.rl

0 directories, 8 files
```

Make errors with

    *** No rule to make target 'unicode.rl', needed by 'scanner.gen.go'.  Stop.

--- a/internal/scanner/Makefile
+++ b/internal/scanner/Makefile
@@ -2,7 +2,7 @@ GO_GENERATE := go generate $(GO_ARGS)
 
 all: scanner.gen.go
 
-scanner.gen.go: scanner.rl unicode.rl unicode2ragel.rb
+scanner.gen.go: scanner.rl unicode2ragel.rb
 	$(GO_GENERATE) -x
 
 clean:

